utils/global-state-update-gen/README.md

```diff
@@ -0,0 +1,106 @@
+```
+
+The `[[accounts]]` definitions control the balances and stakes of accounts on the network. It is possible to change the set of validators using these definitions, by changing the staked amounts.
+
+For every such definition, if the `balance` key is present, the balance of the account will be updated. The account will be created if it didn't exist previously. If the `balance` key is not present, the pre-existing balance (if any) will be left untouched.
+
+Updating the stakes behaves differently based on the value of `only_listed_validators`. If it is false, the existing list of validators is treated as a base, and stakes are modified based on the entries in the config. If the `stake` key is present, the stake is set to the configured value. If it is not present, the pre-existing stake is left untouched.
+
+If `only_listed_validators` is true, pre-existing validators are discarded, and only the accounts with non-zero stakes configured in the config file will be validators after the update. This option exists to match the behavior of the legacy `validators` subcommand and to cater to some use cases in testing.
+
+So, for example, if the network has 100 validators and we want to only change the stake of a single one:
+- with `only_listed_validators` set to false, we need only a single `[[accounts]]` entry for the validator we want to change,
+- with `only_listed_validators` set to true, we need 100 `[[accounts]]` entries, one per each account that is supposed to be a validator after the upgrade.
+
+On the other hand, replacing 100 validators with 5 different ones (a use case in testing setups) would require:
+- just 5 entries for the new validators if `only_listed_validators` is true,
+- 105 entries - 100 to remove the old validators, and 5 to add the new ones - if `only_listed_validators` is false.
+
+The `[[transfers]]` definitions simply transfer funds from one account to another. Every definition requires a source account, a target account and an amount to be defined. If the source account doesn't contain enough funds for the transfer, it won't be executed. If the target account doesn't exist, it will be created.
+
+**Note:** transfers are executed before the `[[accounts]]` definitions. This means that it is possible to overwrite the effects of a transfer if the source or target account is also specified among `[[accounts]]`.
+
+After the transfers have been executed and account balances and stakes have been updated, the tool also updates the auction contract state in the form of the snapshot of validators sets for the next few eras, as well as bids and withdraws:
+- bids of accounts with zero stake are set to empty, and bids of accounts with nonzero stake are created or updated to reflect the configured amounts,
+- if `only_listed_validators` is true, any bid larger than the smallest stake among the new set of validators is reset to zero,
```

Comment:
That means that even bidders who never were validators could get slashed?
E.g. if the lowest validator bid was 100, I bid 90, and the tool changes the validators, so that the lowest stake is 80, my bid will be set to zero?

I wonder whether we should make explicit which bids are supposed to be slashed and which are automatically withdrawn or evicted or something?

Reply:
Yes, unfortunately. Otherwise they'd become validators after the next `step` has been run - which might not be a bad thing, maybe? :thinking: We should probably figure out a more detailed specification of what should happen in these edge cases.